### PR TITLE
fix: ugc Save rocket and car modular assembly data to database

### PR DESCRIPTION
**Pull request by EmosewaMC**
Not sure of a cleaner way to do this besides crawling the full table of user data and searching for subkey x and then further parsing for the modular assembly parts

Tested that the data is inserted into the database in the format of 1:part1+1:part2+1:part3 or for cars up to 7.

#1278 needs to be merged first in either case.

### CODE REVIEW

File: migrations/dlu/12_modular_build_ugc.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE IF NOT EXISTS ugc_modular_build (
+    id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+	ugc_id BIGINT NOT NULL,
+    rocket_ldf VARCHAR(60) NOT NULL
```
Comment (aronwk-aaron) [inline migrations/dlu/12_modular_build_ugc.sql:4]: Rocket ldf? Isn't that supposed to work for cars too. Maybe some better generic name

File: migrations/dlu/12_modular_build_ugc.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE IF NOT EXISTS ugc_modular_build (
+    id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+	ugc_id BIGINT NOT NULL,
```
Comment (aronwk-aaron) [inline migrations/dlu/12_modular_build_ugc.sql:3]: Why is Id and ugc_id needed? Ugc_id should be unique and can serve as a primary key

**Review by aronwk-aaron — CHANGES_REQUESTED** [review]

File: migrations/dlu/12_modular_build_ugc.sql (hunk 1 above)
Comment (EmosewaMC) [reply migrations/dlu/12_modular_build_ugc.sql:4]: I think you can figure out why I named it rocket ldf.

File: migrations/dlu/12_modular_build_ugc.sql [hunk 3]
```diff
@@ -1,5 +1,4 @@
 CREATE TABLE IF NOT EXISTS ugc_modular_build (
```
Comment (aronwk-aaron) [inline migrations/dlu/12_modular_build_ugc.sql:1]: having the character_id as a FK on this table would help remove data when characters are deleted.
sooo something like...

```sql
CREATE TABLE IF NOT EXISTS ugc_modular_build (
	ugc_id BIGINT NOT NULL PRIMARY KEY,
	character_id BIGINT NOT NULL,
	ldf_config VARCHAR(60) NOT NULL,
	FOREIGN_KEY (character_id) REFERENCES charinfo(id) ON DELETE CASCADE,
);
```

Additionally, when entering modular build mode with a vehicle, is this data entry cleaned up? or is it reused somehow?

**Review by aronwk-aaron — APPROVED** [review]

**Review by Jettford — APPROVED** [review]

**Pull request merged**